### PR TITLE
update otel collector to 0.96.0 and trace2receiver to 0.5.2

### PR DESCRIPTION
**Pull request by jeffhostetler**
Update the generated collector code to v0.96.0.
Update the trace2receiver dependency to v0.5.2.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package main
 
 import (
 	"log"
+
```
Comment (mjcheetham) [inline main.go:8]: Generated code/tooling add the new line?

File: main_others.go [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // Code generated by "go.opentelemetry.io/collector/cmd/builder". DO NOT EDIT.
 
 //go:build !windows
-// +build !windows
```
Comment (mjcheetham) [inline main_others.go:4]: What did this comment mean?

**Review by mjcheetham — COMMENTED** [review]
For the code changes in `main_windows.go`, are these related to the updated dependency packages changing their API?

File: main_others.go (hunk 2 above)
Comment (jeffhostetler) [reply main_others.go:4]: I'm not sure.  It's all generated code and I updated the newer template/builder tool and they changed it.

Did GOLANG have multiple tool directives at one point?  I'll look.

Comment (jeffhostetler) [reply main_others.go:4]: https://stackoverflow.com/questions/68360688/whats-the-difference-between-gobuild-and-build-directives

Comment (mjcheetham) [reply main_others.go:4]: Lovely... 😆

**Review by mjcheetham — APPROVED** [review]

**Pull request merged**